Tests/MongoSwiftTests/MongoCollectionTests.swift

```diff
@@ -430,7 +430,7 @@ final class MongoCollectionTests: XCTestCase {
 
         let result2 = try self.coll.insertMany([["_id": nil], ["_id": 20]])
         expect(result2).toNot(beNil())
-        expect(result2?.insertedIds[0]).to(beNil())
+        expect(result2?.insertedIds[0]!).to(beNil())
```

Comment:
Why do you use `!` here but not on the line below? Should they be consistent?

Reply:
This is an interesting thing I discovered yesterday that arises when your dictionary values are an optional type. Basically, a dictionary has an `associatedtype Value`, and the return type of the subscript getter is `Value?`. In the case of a dictionary where `Value` is already an optional type, that means the subscript getter returns a double-optional `??`. So because `insertedIds` is a `[String: BsonValue?]`, the return type of the subscript getter is double-optional, a `BsonValue??`. Essentially, if a key is present but its value is set to nil, you get back `Optional(nil)`. If a key is not present, you get back just plain `nil`. 

So in this line, I'm expecting an `Optional(nil)`, as the key should be present but set to nil.  But `isNil` is actually checking for a true `nil`, and `Optional(nil) != nil`. This is not is not really intuitive, because for all other Swift types, `Optional(x) == x` evaluates to `true`, as is the case in the following line where I compare an `Int?` to an `Int`. 

I alluded to this in [SWIFT-193](https://jira.mongodb.org/browse/SWIFT-193) yesterday as it could be a way to solve our issue of users being unable to differentiate between a key that isn't present and a key that's set to `nil`, though I think it has some other usability tradeoffs. here's an old but still pretty relevant [blog post](https://developer.apple.com/swift/blog/?id=12).